### ISSUE TITLE
API v2 may break ansible deployment due to pagination

### ISSUE THREAD

**Issue by joshbrooks** [onlyhavecans/dnsimple-python#39]
Version 2 of the API introduces pagination to any request returning a list - from [the docs](https://developer.dnsimple.com/v2/#pagination)

> Any API endpoint that returns a list of items requires pagination. By default we will return 30 records from any listing endpoint.

This lead to the dnsimple module in Ansible often failing (wherever there are more than 30 items returned - in our case we have 180+ records) with an error message about zones already existing. 

There's a work around via a hacked version of dnsimple-python at [this repo](https://github.com/catalpainternational/dnsimple-python/commit/968c9ddbca15b4912a4a816c6168c4d55449d67e) which 'depaginates' those requests and renames parameters returned by 'records' to what ansible expects for priority and record type fields.

**Comment by onlyhavecans** [comment]
I think the better solution would be to move handling pagination recursion up into ansible so that we can still try to match the API and official clients better.

**Comment by paklids** [comment]
I can confirm that we hit this bug with just under 400 records. Thank you for the pointer on the work around.

**Comment by glasnt** [comment]
Is there any documentation on how to use the current packaged version (1.1.0) to paginate results? Or is a PR required to allow passing `per_page`/`page` parameters to the API?

**Comment by onlyhavecans** [comment]
I've looked into this more and right now pagination is not usable in 1.1.0

the more I think about this I think for now I'd like to pull in @joshbrooks changes just so this can be less broken.

Copying the official clients will require a lot more of a refactor

**Comment by onlyhavecans** [comment]
I pulled in the changes, made a small tweak during testing and released 1.2.0 if there is any issues let me know.

**Comment by glasnt** [comment]
Thank you @onlyhavecans! 

Pagination now implicitly works for me. 

Using DNSimple==1.1.0
```
>>> from dnsimple import DNSimple
>>> dns = DNSimple()
>>> len(dns.records(dns.domain('<testingdomain>')['domain']['id']))
30
```

Using DNSimple==1.2.0
```
>>> from dnsimple import DNSimple
>>> dns = DNSimple()
>>> len(dns.records(dns.domain('<testingdomain>')['domain']['id']))
131
```

💃
